plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/FlightStreamPlugin.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.arrow.flight;
+
+import org.opensearch.arrow.flight.bootstrap.FlightStreamPluginImpl;
+import org.opensearch.arrow.spi.StreamManager;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.node.DiscoveryNodes;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.network.NetworkService;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.IndexScopedSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.settings.SettingsFilter;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.common.util.PageCacheRecycler;
+import org.opensearch.core.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.core.indices.breaker.CircuitBreakerService;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.env.Environment;
+import org.opensearch.env.NodeEnvironment;
+import org.opensearch.plugins.SecureTransportSettingsProvider;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.rest.RestController;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.script.ScriptService;
+import org.opensearch.telemetry.tracing.Tracer;
+import org.opensearch.threadpool.ExecutorBuilder;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.Transport;
+import org.opensearch.transport.client.Client;
+import org.opensearch.watcher.ResourceWatcherService;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.opensearch.common.util.FeatureFlags.ARROW_STREAMS_SETTING;
+
+/**
+ * Delegates the plugin implementation to the {@link FlightStreamPluginImpl} if the Arrow Streams feature flag is enabled.
+ * Otherwise, it creates a no-op implementation.
+ */
+@ExperimentalApi
+public class FlightStreamPlugin extends BaseFlightStreamPlugin {
+
+    private final BaseFlightStreamPlugin delegate;
+
+    /**
+     * Constructor for FlightStreamPlugin.
+     * @param settings The settings for the plugin.
+     */
+    public FlightStreamPlugin(Settings settings) {
+        if (FeatureFlags.isEnabled(ARROW_STREAMS_SETTING)) {
```

Comment:
I don't think you need to have 2 implementations of the plugin, if `FeatureFlags` is not set - just don't register any transports at all

Reply:
I felt this is a safer approach given its just not transport but also exposing other things. This way any developer working on this plugin won't accidentally end up registering something unwanted tomorrow if feature flag is disabled as its easy to miss adding feature flag check.

Reply:
If you want to be safe, just throw an exception on initialization - onStart / constructor - so the service would refuse to start, we don't need wrappers for that I think.

Reply:
other than flightService, it also adds stuff like executorBuilders, Actions, various settings and have logic onNodeStarted(). 
I do not want any of it to be initialized if feature flag is disabled. I think this is a safer approach and cleaner approach than adding checks for feature flags at various places, there could be more stuff this plugin initializes tomorrow.

Reply:
To be fair, it adds tons of unnecessary code to support, if plugin is installed but feature flag is not set - throw exception on construction so the service won't start

Reply:
done